### PR TITLE
Moves TimeValue setting to java

### LINKED ISSUE #18757: Reimplement LogStash::Setting::TimeValue to Java

### PR DESCRIPTION
**Pull request by andsel**
## Release notes
[rn:skip]

## What does this PR do?
Moves the Ruby setting class TimeValue setting to Java. It covers the test in unit test while keeping and adapting the original time_value_spec.rb test suite to prove it still works on Ruby code.

## Why is it important/What is the impact to the user?

N/A

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist
- [x] Test using the setting on Logstash run.

## How to test this PR locally
Change a setting that leverage this implementation and check it works as expected.
1. create Logstash `pipeline.conf` file:

      ```
      input {
        generator {
          message => "Hello world!"
        }
      }
      
      output {
        sink {}
      }
      ```

2. Update the reload interval setting in `config/logstash.yml` as:

      ```yaml
      config.reload.automatic: true
      config.reload.interval: 30s
      ```
      
3. run Logstash       

      ```sh
      bin/logstash -f /path/to/pipeline.conf
      ```

4. Update the pipeline.conf file, adding a comment for example, and verify in the logs the interval in pipeline reload is greater than 30 seconds.
    
## Related issues

- Closes #18757

## Logs

```
[2026-02-10T10:41:28,935][INFO ][logstash.javapipeline    ][main] Pipeline `main` is configured with `pipeline.ecs_compatibility: v8` setting. All plugins in this pipeline will default to `ecs_compatibility => v8` unless explicitly configured otherwise.
[2026-02-10T10:41:28,941][INFO ][logstash.javapipeline    ][main] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>12, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>50, "pipeline.max_inflight"=>1500, "batch_metric_sampling"=>"minimal", "pipeline.sources"=>["/Users/andrea/workspace/logstash_andsel/simple_generator_pipeline2.conf"], :thread=>"#<Thread:0x88a7390 /Users/andrea/workspace/logstash_andsel/logstash-core/lib/logstash/java_pipeline.rb:147 run>"}
[2026-02-10T10:41:28,955][INFO ][logstash.javapipeline    ][main] Pipeline Java execution initialization time {"seconds"=>0.01}
[2026-02-10T10:41:28,970][INFO ][logstash.javapipeline    ][main] Pipeline started {"pipeline.id"=>"main"}
[2026-02-10T10:41:28,978][INFO ][logstash.agent           ] Pipelines running {:count=>1, :running_pipelines=>[:main], :non_running_pipelines=>[]}


[2026-02-10T10:41:57,888][INFO ][logstash.pipelineaction.reload] Reloading pipeline {"pipeline.id"=>:main}
[2026-02-10T10:41:58,054][INFO ][logstash.javapipeline    ][main] Pipeline terminated {"pipeline.id"=>"main"}
```

### CODE REVIEW

**Comment by andsel** [comment]
run exhaustive tests

File: logstash-core/spec/logstash/settings/time_value_spec.rb [hunk 1]
```diff
@@ -40,16 +40,10 @@
...
     context "when a value is given as a nanosecond" do
-      let(:deprecation_logger_stub) { double("DeprecationLogger").as_null_object }
-      before(:each) do
-        allow(subject).to receive(:deprecation_logger).and_return(deprecation_logger_stub)
-      end
       it "should set the value" do
         subject.set(5)
         expect(subject.value).to eq(LogStash::Util::TimeValue.new(5, :nanosecond))
         expect(subject.value.to_nanos).to eq(5)
-
```
Comment (andsel) [inline logstash-core/spec/logstash/settings/time_value_spec.rb:51]: **Note for reviewer**
The verification of deprecation logger switched implementation. However, this case is covered by the JUnit test case

File: logstash-core/spec/logstash/settings/time_value_spec.rb [hunk 2]
```diff
@@ -40,16 +40,10 @@
...
-        expect(deprecation_logger_stub).to have_received(:deprecated).with(/units will be required/).once
       end
```
[the remark this answers is not in the document]
Comment (andsel) [reply logstash-core/spec/logstash/settings/time_value_spec.rb:47]: This is already covered in the JUnit twin test

File: logstash-core/lib/logstash/settings.rb [hunk 3]
```diff
@@ -455,25 +455,7 @@ def validate(value)
...
-                                        "Time units will be required in a future release of Logstash. " +
-                                        "Acceptable unit suffixes are: `d`, `h`, `m`, `s`, `ms`, `micros`, and `nanos`.")
-
-          return Util::TimeValue.new(value, :nanosecond)
-        end
-
-        Util::TimeValue.from_value(value)
-      end
-    end
+    java_import org.logstash.settings.TimeValueSetting
```
[the remark this answers is not in the document]
Comment (andsel) [reply logstash-core/lib/logstash/settings.rb:459]: No need for this compatibility alias because:
- it's an internal API
- all instances were updated

File: logstash-core/src/main/java/org/logstash/settings/TimeValueSetting.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        return coerceStatic(getName(), value);
+    }
+
+    private static TimeValue coerceStatic(String name, Object value) {
+        if (value instanceof Number) {
+            DEPRECATION_LOGGER.deprecated(
+                "Integer value for `" + name + "` does not have a time unit and will be interpreted in nanoseconds. " +
+                "Time units will be required in a future release of Logstash. " +
+                "Acceptable unit suffixes are: `d`, `h`, `m`, `s`, `ms`, `micros`, and `nanos`.");
+            return new TimeValue(((Number) value).intValue(), "nanosecond");
```
[the remark this answers is not in the document]
Comment (andsel) [reply logstash-core/src/main/java/org/logstash/settings/TimeValueSetting.java]: Fixed with a4a5eb20440bcb81218176e0a42098454c8cd82e

File: logstash-core/spec/logstash/settings/time_value_spec.rb [hunk 5]
```diff
@@ -40,16 +40,10 @@
     end
 
     context "when a value is given as a nanosecond" do
-      let(:deprecation_logger_stub) { double("DeprecationLogger").as_null_object }
-      before(:each) do
-        allow(subject).to receive(:deprecation_logger).and_return(deprecation_logger_stub)
-      end
       it "should set the value" do
```
Comment (jsvd) [inline logstash-core/spec/logstash/settings/time_value_spec.rb:43]: In code migrations like this, it is important to test edge cases. for example a high number enough number should throw a range error:

Logstash 9.2.0:

```
3.1.0 :007 > setting = LogStash::Setting::TimeValue.new("test", "3s")
 => <LogStash::Setting::TimeValue(test): #<Java::OrgLogstashUtil::TimeValue:0x3132e390> (DEFAULT)>
3.1.0 :008 > setting.set(250000000); setting.value.to_nanos
 => 250000000
3.1.0 :009 > setting.set(2500000000); setting.value.to_nanos
org/jruby/RubyClass.java:936:in `new': too big for int: 2500000000 (RangeError)
        from /private/tmp/logstash-9.2.0/logstash-core/lib/logstash/settings.rb:624:in `coerce'
        from /private/tmp/logstash-9.2.0/logstash-core/lib/logstash/settings.rb:405:in `set'
        from (irb):9:in `evaluate'
```

In this PR it will overflow and return a negative number:

```
jruby-9.4.13.0 :011 > setting = LogStash::Setting::TimeValueSetting.new("test", "3s")
 => #<Java::OrgLogstashSettings::TimeValueSetting:0xe6093da>
jruby-9.4.13.0 :012 > setting.set(250000000); setting.value.to_nanos
[2026-02-25T11:50:23,278][WARN ][org.logstash.deprecation.settings.TimeValueSetting] Integer value for `test` does not have a time unit and will be interpreted in nanoseconds. Time units will be required in a future release of Logstash. Acceptable unit suffixes are: `d`, `h`, `m`, `s`, `ms`, `micros`, and `nanos`.
 => 250000000
jruby-9.4.13.0 :013 > setting.set(2500000000); setting.value.to_nanos
[2026-02-25T11:50:26,895][WARN ][org.logstash.deprecation.settings.TimeValueSetting] Integer value for `test` does not have a time unit and will be interpreted in nanoseconds. Time units will be required in a future release of Logstash. Acceptable unit suffixes are: `d`, `h`, `m`, `s`, `ms`, `micros`, and `nanos`.
 => -1794967296
```

While unlikely and wouldn't happend during an upgrade, someone setting this too high to disable, for example, auto reload interval (we've seen users do it), will cause `converge_state_and_update` to be called in a tight loop, heavily increasing CPU usage.
